cmd/mole/main.go

```diff
@@ -49,15 +57,82 @@ func main() {
 		if err != nil {
 			os.Exit(1)
 		}
+	case "stop":
+		err := stopDaemon(*app)
+		if err != nil {
+			os.Exit(1)
+		}
 	case "aliases":
 		err := lsAliases(*app)
 		if err != nil {
 			os.Exit(1)
 		}
+}
+
+func stopDaemon(app cli.App) error {
+	workDir := fmt.Sprintf("%s/.mole/instances/%s", os.Getenv("HOME"), app.Stop)
+	pidPathName := fmt.Sprintf("%s/pid", workDir)
+	logPathName := fmt.Sprintf("%s/log", workDir)
+
+	cntxt := &daemon.Context{
+		PidFileName: pidPathName,
+		PidFilePerm: 0644,
+		LogFileName: logPathName,
+		LogFilePerm: 0640,
+		Umask:       027,
+		Args:        os.Args,
+	}
+	d, err := cntxt.Search()
+	if err != nil {
+		log.Fatalln("Unable send signal to the daemon:", err)
+  }
+  d.Kill()
+	return nil
+}
+
+func createInstancesDir() {
+	moleDir := fmt.Sprintf("%s/.mole", os.Getenv("HOME"))
+	if _, err := os.Stat(moleDir); os.IsNotExist(err) {
+		os.Mkdir(moleDir, 0755)
+	}
+	instancesDir := fmt.Sprintf("%s/instances", moleDir)
+	if _, err := os.Stat(instancesDir); os.IsNotExist(err) {
+		os.Mkdir(instancesDir, 0755)
 	}
 }
 
 func startFromAlias(app cli.App) error {
+	if app.Detach {
+		cntxt := &daemon.Context{
```

Comment:
Why not preparing the instance file system to create the daemon context pointing to the right location?

Reply:
in go-daemon, i cannot change location of create pid & mole.log
It's always create at "./"
so I use os.Rename() to move file to correct path to fix this problem